client/state/imports/site-importer/actions.js

```diff
@@ -182,32 +183,60 @@ export const validateSiteIsImportable =
 		);
 		dispatch( startSiteImporterIsSiteImportable() );
 
-		return wpcom.req
-			.get( {
-				path: `/sites/${ siteId }/site-importer/is-site-importable?${ stringify( params ) }`,
-				apiNamespace: 'wpcom/v2',
-			} )
-			.then( ( response ) => {
-				dispatch(
-					recordTracksEvent( 'calypso_site_importer_validate_site_success', {
-						blog_id: siteId,
-						site_url: response.site_url,
-						supported_content: sortAndStringify( response.supported_content ),
-						unsupported_content: sortAndStringify( response.unsupported_content ),
-						site_engine: response.engine,
-					} )
-				);
-				dispatch( siteImporterIsSiteImportableSuccessful( response ) );
-			} )
-			.catch( ( error ) => {
+		const isSiteImportable = wpcom.req.get( {
+			path: `/sites/${ siteId }/site-importer/is-site-importable?${ stringify( params ) }`,
+			apiNamespace: 'wpcom/v2',
+		} );
+
+		const analyzeUrl = wpcom.req.get( {
+			path: `/imports/analyze-url?${ stringify( { site_url: targetSiteUrl } ) }`,
+			apiNamespace: 'wpcom/v2',
+		} );
+
+		const [ isSiteImportableResult, analyzeUrlResult ] = await Promise.allSettled( [
+			isSiteImportable,
+			analyzeUrl,
+		] );
+
+		if ( isSiteImportableResult.status === 'fulfilled' ) {
+			const response = isSiteImportableResult.value;
+			dispatch(
+				recordTracksEvent( 'calypso_site_importer_validate_site_success', {
+					blog_id: siteId,
+					site_url: response.site_url,
+					supported_content: sortAndStringify( response.supported_content ),
+					unsupported_content: sortAndStringify( response.unsupported_content ),
+					site_engine: response.engine,
+				} )
+			);
+			dispatch( siteImporterIsSiteImportableSuccessful( response ) );
+		} else {
+			dispatch(
+				recordTracksEvent( 'calypso_site_importer_validate_site_fail', {
+					blog_id: siteId,
+					site_url: targetSiteUrl,
+				} )
+			);
+
+			// do platform validation if param - targetPlatform is given
+			if (
+				isSiteImportableResult?.reason?.code === 1000002 &&
+				targetPlatform &&
+				analyzeUrlResult?.value?.platform !== targetPlatform
+			) {
 				dispatch(
-					recordTracksEvent( 'calypso_site_importer_validate_site_fail', {
-						blog_id: siteId,
-						site_url: targetSiteUrl,
+					siteImporterIsSiteImportableFailed( {
+						message: `The URL you entered does not seem to be a ${ convertPlatformName(
+							targetPlatform
+						) } site.`,
```

Comment:
This message must be translated. You can find an example here: [client/blocks/import/ready/index.tsx#L63-L72](https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/import/ready/index.tsx#L63-L72)

Reply:
Thanks for the reminder. I find the message will be translated at https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/importer/error-pane.jsx#L29.

Reply:
No, it will not. That phrase is:
`%(errorDescription)s{{br/}}{{a}}Try again{{/a}} or {{cs}}get help{{/cs}}`
But `errorDescription` will be replaced by an English phrase because you haven't translated it _before_.

So for example, in Italian:
`%(errorDescription)s{{br/}}{{a}}Try again{{/a}} or {{cs}}get help{{/cs}}.`
`%(errorDescription)s{{br/}}{{a}}Prova di nuovo{{/a}} o {{cs}}chiedi aiuto{{/cs}}.`

At runtime the output will be:
`The URL you entered does not seem to be a Wix site{{br/}}{{a}}Prova di nuovo{{/a}} o {{cs}}chiedi aiuto{{/cs}}.`

The translators see the raw strings, not the runtime version, where all the values are expanded.

Also is better to not have _functions_ as interpolated values but to use a variable with a name that will help the translators.

Does it make sense to you?

Reply:
So, you mean we should translate the string this way https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/import/ready/index.tsx#L63-L72 instead of https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/importer/error-pane.jsx#L29-L41? The [i18n-calypso document](https://github.com/Automattic/wp-calypso/tree/trunk/packages/i18n-calypso#string-substitution) says "The translate() method uses sprintf interpolation for string substitution", which is very similar to the approach you suggest.   Just wondering what's the differences between these two approaches and what's the best way to do localization?

Reply:
Using `translate` is the best way. But what I'm saying here is that everything that is interpolated inside an i18n string must be localized before otherwise, it will remain in English. 

The automatic tool that generate .po files extract this string:

`%(errorDescription)s{{br/}}{{a}}Try again{{/a}} or {{cs}}get help{{/cs}}`

That will be translated and at runtime React will substitute `errorDescription` with your `The URL you entered does...` phrase. If that string is not translated then it will remain in English.

Reply:
Thank you for the detailed explanation. So... in other words, does it means that this bug has existed for a long time? In fact, this PR doesn't modify the original way to translate strings but only alters the error string we dispatch to the Redux store.

Reply:
Hello @zaerl, the error message translation is fixed https://github.com/Automattic/wp-calypso/pull/64881/commits/fe14701c9225c3cda97024deb8f55c29fd308735